lib/pubsub_client/subscriber.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'google/cloud/pubsub'
+
+module PubsubClient
+  class Subscriber
+    DEFAULT_CONCURRENCY = 8
+
+    # @param subscription [Google::Cloud::PubSub::Subscription]
+    def initialize(subscription)
+      @subscription = subscription
+    end
+
+    # @param concurrency [Integer]
+    # @param auto_ack [Boolean]
+    def subscribe(concurrency, auto_ack, &block)
+      listener = @subscription.listen(threads: { callback: concurrency }) do |received_message|
+        yield received_message.data
+        received_message.acknowledge! if auto_ack
+      end
+
+      begin
+        listener.start
```

Comment:
The [example implementation in Google's documentation](https://github.com/googleapis/google-cloud-ruby/tree/master/google-cloud-pubsub#example) shows there is an error handler (`listener.on_error`). You should expose this so clients can get alerted if there are any errors.